### PR TITLE
Add basis_set_exchange

### PR DESCRIPTION
**Pull request by mattwthompson**
cc: @bennybp - please comment that you agree to be a maintainer. Also, any idea why the license file is not in the tarball on PyPI?

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by mattwthompson** [comment]
I've removed the license file for now, just to get things built. I understand the reasons why it should be packaged and will make sure it's added in during the next release

**Comment by bennybp** [comment]
I just added it to the BSE setup.py. I can create a new version by the end of the week (which I have been meaning to do for a while)

**Comment by bennybp** [comment]
Basis set exchange v0.8.13 is released and on PyPI. This includes the LICENSE file

**Comment by mattwthompson** [comment]
Thanks @bennybp! I do need one more thing, either

1. Agree to be a maintainer by commenting something like "I agree to be a maintainer of this recipe" here in this PR, or
2. Don't agree to, if you don't wish to. Just be aware that I'd be the sole authority (outside of the conda-forge core team) on pushing changes to this repo. Happy to include anybody else that would like to share that burden, at MolSSI or otherwise

**Comment by bennybp** [comment]
Oh yeah, I forgot about that. Yes, I do agree to be the maintainer of this recipe

**Comment by mattwthompson** [comment]
I think this is ready for review @conda-forge/help-python - my understanding is that the Windows failure is acceptable since this is a pure Python package and won't actually be built cross-platform (https://github.com/conda-forge/staged-recipes/pull/12129#issuecomment-671399302) although I'm not sure if we're able to turn that off in CI.

File: recipes/basis_set_exchange/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 78e2cf3d9b6862e82b27e019c09812e01d40b785014a3969fd51b52079c90122
+
+build:
+  noarch: python
+  number: 0
+  script: "python -m pip install . -vv"
```
Comment (synapticarbors) [inline recipes/basis_set_exchange/meta.yaml:15]: use canonical form in `example/meta.yaml`:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L35

`script: "{{ PYTHON }} -m pip install . -vv"`

Also it looks like this project has python entry points that should be listed:

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#python-entry-points

and then tested under the `tests` section:

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-commands

File: recipes/basis_set_exchange/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+requirements:
+  host:
+    - python
```
Comment (synapticarbors) [inline recipes/basis_set_exchange/meta.yaml:19]: Make `python` -> `python >=3.5` under `host` and `run`.

**Review by synapticarbors — CHANGES_REQUESTED** [review]

File: recipes/basis_set_exchange/meta.yaml (hunk 2 above)
Comment (mattwthompson) [reply recipes/basis_set_exchange/meta.yaml]: Done

File: recipes/basis_set_exchange/meta.yaml (hunk 1 above)
Comment (mattwthompson) [reply recipes/basis_set_exchange/meta.yaml]: I added the entry points to the build section and added `cmd -h` to the testing block, which is my reading of the docs

**Review by synapticarbors — APPROVED** [review]

**Pull request merged**